src/uu/chroot/src/chroot.rs

```diff
@@ -10,50 +10,61 @@
 
 #[macro_use]
 extern crate uucore;
-use uucore::entries;
-use uucore::libc::{self, chroot, setgid, setgroups, setuid};
-
+use clap::{App, Arg};
 use std::ffi::CString;
 use std::io::Error;
 use std::path::Path;
 use std::process::Command;
+use uucore::entries;
+use uucore::libc::{self, chroot, setgid, setgroups, setuid};
 
+static VERSION: &str = env!("CARGO_PKG_VERSION");
 static NAME: &str = "chroot";
+static ABOUT: &str = "Run COMMAND with root directory set to NEWROOT.";
 static SYNTAX: &str = "[OPTION]... NEWROOT [COMMAND [ARG]...]";
-static SUMMARY: &str = "Run COMMAND with root directory set to NEWROOT.";
-static LONG_HELP: &str = "
- If COMMAND is not specified, it defaults to '$(SHELL) -i'.
- If $(SHELL) is not set, /bin/sh is used.
-";
 
 pub fn uumain(args: impl uucore::Args) -> i32 {
     let args = args.collect_str();
 
-    let matches = app!(SYNTAX, SUMMARY, LONG_HELP)
-        .optopt(
-            "u",
-            "user",
-            "User (ID or name) to switch before running the program",
-            "USER",
+    let matches = App::new(executable!())
+        .version(VERSION)
+        .about(ABOUT)
+        .usage(SYNTAX)
+        .arg(Arg::with_name("newroot").hidden(true))
+        .arg(
+            Arg::with_name("user")
+                .short("u")
+                .long("user")
+                .help("User (ID or name) to switch before running the program")
+                .value_name("USER"),
         )
-        .optopt("g", "group", "Group (ID or name) to switch to", "GROUP")
-        .optopt(
-            "G",
-            "groups",
-            "Comma-separated list of groups to switch to",
-            "GROUP1,GROUP2...",
+        .arg(
+            Arg::with_name("group")
+                .short("g")
+                .long("group")
+                .help("Group (ID or name) to switch to")
+                .value_name("GROUP"),
         )
-        .optopt(
-            "",
-            "userspec",
-            "Colon-separated user and group to switch to. \
+        .arg(
+            Arg::with_name("groups")
+                .short("G")
+                .long("groups")
+                .help("Comma-separated list of groups to switch to")
+                .value_name("GROUP1,GROUP2..."),
+        )
+        .arg(
+            Arg::with_name("userspec")
+                .long("userspec")
+                .help(
+                    "Colon-separated user and group to switch to. \
              Same as -u USER -g GROUP. \
              Userspec has higher preference than -u and/or -g",
-            "USER:GROUP",
+                )
+                .value_name("USER:GROUP"),
         )
-        .parse(args);
+        .get_matches_from(args);
 
-    if matches.free.is_empty() {
+    if matches.args.is_empty() {
```

Comment:
I think you can enforce that above: make sure that clap verifies that it isn't empty

Reply:
maybe like
https://github.com/uutils/coreutils/blob/master/src/uu/df/src/df.rs#L850-L857
?

Reply:
did you try this?
it should simplify the code

Reply:
I believe it is needed in his case since he uses `multiple(true)`.

Reply:
You have min_values for this :)

example: https://github.com/uutils/coreutils/blob/a42e24e088a0ffa38af6549caa304c0cfcf4ac28/src/uu/realpath/src/realpath.rs#L61

Reply:
it didn't work?

Reply:
TBH, i wasn't sure which flag needs minimum value or multiple value, we use multiple values only for userspec and groups but they are seperated by colon or comma. Can you point me to the right direction if i'm missing smth :)